### ISSUE TITLE
GetVertices and GetIndices() out of order

### ISSUE THREAD

**Issue by ZanderCode** [Facepunch/sbox-issues#6256]
### Describe the bug

I have a rectangular mesh with numbered verticies (blender)
![image](https://github.com/user-attachments/assets/076efee1-7c53-4e39-a348-188bc1a4d302)
when I call the GetVertices() and GetIndices() functions of the ModelRenderer.Model class, I get out of order vertices.

I loop through the indices and I try to trace the generation of triangles and somehow the vertex of the third triangle references the wrong index. Output from console for the GetIndces() function:

 02:56:05   GameMenu 0
 02:56:05   GameMenu 1
 02:56:05   GameMenu 2

 02:56:05   GameMenu 0
 02:56:05   GameMenu 2
 02:56:05   GameMenu 3

 02:56:05   GameMenu 2 <-  (no edge connects these points)
 02:56:05   GameMenu 1 <- (no edge connects these points)
 02:56:05   GameMenu 4

... Also another thing I noticed when instancing gameobjects along the vertices was that the vertices were not in order. So you could imagine I want to spawn some sort of way point navigation system procedurally along this mesh, but since the vertex and indicies are all messed up, I can't image how I would do this without hand placing waypoints.

Last notes and thoughts: Maybe the indices list only connects two indices together to form an edge and the list is processed by edges? I could work with this, but it doesn't explain why when I instance things on the vertices the instances end up jumping from the beginning of the mesh to the end and places in-between - instead of tracing the vertices as they are laid out in the image

### To Reproduce

1. Create a rectangular mesh in blender with subdivisions along the long axis
2. export as fbx and create model in s&box ModelDoc editor
3. add to the model render mesh list - a "RenderMeshMarkup" and tick the CPU vertex access option, then compile
4. Create script that references gameobject with ModelRenderer component.
5. Use the .Model.GetIndices() function of the referenced ModelRenderer.
6. Examine Indices

### Expected behavior

Indices are ordered the way the model is imported

**Comment by MD485** [comment]
It might help if you posted the fbx, an obj, or a .blend file with a model that exhibits this behaviour.
If someone doesn't know exactly what you did in blender then they might not be able to reproduce your issue.

**Comment by WheatleyMF** [comment]
Yeah last time I've messed with GetVertices() and GetIndices() I've also noticed that sometimes output is inconsistent. But code repro would definitely be useful here

**Comment by ZanderCode** [comment]
Build a mesh in 3D software like how I have above and run this:

```
[Property] public ModelRenderer modelRenderer;

protected override void OnStart(){
    int[] indices = modelRenderer.Model.GetIndices();
    foreach (int indx in indices){
       Info.Log(indx);
   }
}
```
Examine how every two indices printed represent an edge connection. Before, I suspected triangle formations (so every three indices were a triangle), but it seems that every two indices is an edge.

I wonder if s&box will chunk the mesh in an optimized way. Maybe it finds and orders edges to render based on height or something. I have no clue, but the image I show above has ordered indices and s&box does not maintain the order.

**Comment by ZanderCode** [comment]
> It might help if you posted the fbx, an obj, or a .blend file with a model that exhibits this behaviour. If someone doesn't know exactly what you did in blender then they might not be able to reproduce your issue.
> 
> ![image](https://private-user-images.githubusercontent.com/36082767/358942735-4718b163-5a96-4522-a4a2-ae2e517e3997.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwODU1OTIsIm5iZiI6MTcyNDA4NTI5MiwicGF0aCI6Ii8zNjA4Mjc2Ny8zNTg5NDI3MzUtNDcxOGIxNjMtNWE5Ni00NTIyLWE0YTItYWUyZTUxN2UzOTk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE5VDE2MzQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyNGQ5NDlkMTc3Yzg2MTRkY2UxMmYzODFjOTRjNWIxYjcyOWU1NmRjMWYyMzg0Y2Y2Y2YwNzRmZTBkYzFhMTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ux8c5ZbgSD87vXZyukuXFYy5UGOCQhb60HZjMmwdEME) This minor visual darkening near the zero implies the mesh may have strange topology, which wouldn't be a fault of the engine.

I checked with the 3D artist and they said the visual is an artifact of the procedural UV generation and not the placement / ordering of the vertices themselves. I can test with other meshes too

**Comment by aylaylay** [comment]
You didn't give a repro project so I had to guess. See if it's fixed when the game next updates

**Comment by ZanderCode** [comment]
Oh jeez I think I got mixed up as I kept reading repro as steps, not the actual project. I'll try to find some time to get y'all a test project hopefully soon.

**Comment by aylaylay** [comment]
Is this working now?

**Comment by ZanderCode** [comment]
I don't think so. I'm observing what looks like proper indices, but not vertices. Their positions are off by a lot.

**Comment by aylaylay** [comment]
How are they off?

**Comment by ZanderCode** [comment]
It appears that the positions look transformed by multiple thousand units. I've got limited time to post a test project. Hopefully tomorrow I can give an example.

**Comment by aylaylay** [comment]
You could just send the model

**Comment by ZanderCode** [comment]
https://drive.google.com/file/d/1yPW0onC_8oAwB7SJodsM4oMGtSQHhN1V/view?usp=sharing

**Comment by aylaylay** [comment]
![sbox-dev_FCrHccA8I3](https://github.com/user-attachments/assets/c834012d-a80c-496c-91a5-5e088f9444b1)

Works completely fine?

**Comment by aylaylay** [comment]
I don't think it's wise to rely on a compiled render mesh having the same triangulation from blender to fbx to modeldoc
